keras/backend/mxnet_backend.py

```diff
@@ -3042,13 +3042,19 @@ def conv1d(x, kernel, strides=1, padding='valid',
         data_format = image_data_format()
     _validate_data_format(data_format)
 
+    if padding not in {'same', 'valid', 'causal'}:
+        raise ValueError('MXNet Backend: `padding` should be either `same`, `valid` or `causal`.')
+
     # Causal requires temporal padding.
     # MXNet backend does not support temporal padding on 3D tensor.
-    if padding is 'causal':
-        raise ValueError('MXNet Backend: conv1d does not support "causal" padding mode')
+    assert ndim(x) == 3
```

Comment:
Please add error message about the details in case this assert fails

Reply:
Added